index.d.ts

```diff
@@ -75,6 +75,30 @@ declare const isStream: {
 	```
 	*/
 	transform(input: unknown): input is stream.Transform;
+
+	/**
+	@returns Returns a `string` to tell the [`type of stream`](https://nodejs.org/api/stream.html#stream_types_of_streams): `readable`, `writable`, `duplex`, `transform` or `unknown` for unknown stream type and `undefined` for non streams.
+
+	@example
+	```
+	import * as fs from 'fs';
+	import Stream from 'stream';
+	import isStream = require('is-stream');
+
+	isStream.type(fs.createReadStream('unicorn.png'));
+	//=> readable
+
+	isStream.type(fs.createWriteStream('unicorn.png'));
+	//=> writable
+
+	isStream.type({});
+	//=> undefined
+
+	isStream.type(new Stream.Stream());
+	//=> unknown
+	```
+	*/
+	type(input: unknown): input is any;
```

Comment:
The return type should be a union of the possible values.

Reply:
Like this ? 

```js
// index.d.ts
type(input: stream.Stream|any): input is stream.Stream|any;

// index.test-d.ts
expectType<stream.Stream|any>(foo);
```

Reply:
No. I suggest reading a little bit up on TypeScript and union return types :)